airbyte-integrations/connectors/source-file/source_file/source.py

```diff
@@ -67,7 +67,7 @@ class SourceFile(Source):
     content of directories, etc in a latter iteration.
     """
 
-    client_class = Client
+    client_class = ArchiveClient
```

Comment:
This is a bit misleading, at first read it suggests that all the file are considered to be archive. I'd prefer to have a kind of factory method (`_get_client`?)  which creates archive client or normal file client.